spec.md

```diff
@@ -0,0 +1,948 @@
+Implementations MAY support additional encodings.
+Names of encodings are compared in a case-insensitive manner.
+
+> [!IMPORTANT]
+>
+> Many implementations only apply the specified encoding to **subsequent** headers and note texts.
+> Although this is technically not spec-compliant it is usually best to put the `ENCODING` header first.
+
+> [!WARNING]
+>
+> The use of the `ENCODING` tag is highly discouraged.
+> Songs must always use the UTF-8 encoding.
+
+### 3.30. The `RELATIVE` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       "yes" / "no"
+Since:         0.2.0
+Deprecated:    0.3.0
+Removed:       1.0.0
+```
+
+The `RELATIVE` header enables [Relative Mode](#a-relative-mode) (see Appendix A).
+
+## 3. The File Body
+
+The body of a file consists of a sequence of notes, end-of-phrase markers, and voice changes.
+
+```abnf
+body = *( note /
+          end-of-phrase /
+          voice-change /
+          empty-line )
+```
+The sequence of notes and end-of-phrase markers SHOULD appear in ascending order by their start beats.
+
+> [!CAUTION]
+>
+> Whether the body of a file must or may be sorted is not decided yet.
+
+### 3.1. Notes
+
+A note is a musical element in a song.
+Each note is defined by its type, start beat, duration, pitch, and text.
+
+```abnf
+note = note-type
+       WSP start-beat
+       WSP duration
+       WSP pitch
+       WSP note-text
+       line-break
+
+note-type  = %x21-22 / %x24-7E  ; Visible ASCII-characters except space and #
+start-beat = *DIGIT
+duration   = *DIGIT
+pitch      = [ minus ] *DIGIT
+note-text  = 1*( %x20-10FFFF )
+
+minus   = %x2D  ; -
+```
+
+> [!CAUTION]
+>
+> Whether only a single or multiple whitespace character in a row are allowed is currently up for discussion ([#46](https://github.com/UltraStar-Deluxe/format/issues/46)).
+
+The note type indicates how singing the correct or wrong note should affect scoring.
+The following sections define standard note types.
+Implementations MAY substitute unknown note types with freestyle notes (`F`).
+Implementations MUST NOT attach semantics to note types not covered by this specification.
+
+The start beat and duration define the time when a note appears in a song.
+Both are indicated in beats (see section 3.3.) relative to offset indicated by the `GAP` header.
+The end beat of a note is calculated as its start beat plus its duration.
+Notes SHOULD NOT overlap, i.e. the start beat of a note being between the start beat (inclusive) and end beat (exclusive) of another note.
+
+> [!CAUTION]
+>
+> Whether and how applications may define custom note types hasn't been decided yet.
+
+> [!CAUTION]
+>
+> Whether negative note starts and/or durations are valid is, hasn't been decided yet.
+
+The pitch of a note is encoded as the number of half-steps relative to C4 (also referred to as middle C).
+So a pitch of `5` represent an F4 and a pitch of `-2` represents an A#3.
```

Comment:
Please also change the notes to `F4` and `A#3`

Reply:
Fixed (I think). I also added a note about the pitch notation used.